### PR TITLE
ROM for RK2Avg solver

### PR DESCRIPTION
**Pull request by dylan-copeland**
This PR implements ROM for the RK2Avg solver.

### CODE REVIEW

File: laghos_rom.cpp [hunk 1]
```diff
@@ -48,6 +52,10 @@ void ROM_Sampler::SampleSolution(const double t, const double dt, Vector const&
 
         generator_V->takeSample(V.GetData(), t, dt);
         generator_V->computeNextSampleTime(V.GetData(), dVdt.GetData(), t);
+
+        // Without this check, libROM may use multiple time intervals, and without appropriate implementation
+        // the basis will be from just one interval, resulting in large errors and difficulty in debugging.
+        MFEM_VERIFY(generator_X->getNumBasisTimeIntervals() == 1, "Only 1 basis time interval allowed");
```
Comment (chldkdtn) [inline laghos_rom.cpp:58]: Although this is problematic for this particular simulation, this capability of libROM can be useful for dividing time window by the number of snapshots collected.

**Review by chldkdtn — COMMENTED** [review]
Shouldn't we make "max_model_dim" in laghos_rom.hpp file to be available as a user option to avoid compilation level change?

File: laghos_rom.cpp (hunk 1 above)
Comment (dylan-copeland) [reply laghos_rom.cpp:58]: I agree, but in order to use that capability there needs to be implementation in Laghos, which is not done. In the future we may want to use multiple libROM time intervals, but until that is supported in Laghos we need to warn the user that the basis comes from just one interval and may be inaccurate if misused for the entire simulation.

Comment (chldkdtn) [reply laghos_rom.cpp:58]: Sounds good, Dylan. I will approve and merge it to rom-dev. Kevin is not supposed to work on ROM this week.

**Review by chldkdtn — APPROVED** [review]

**Pull request merged**